### PR TITLE
cmd/openshift-sdn: Allow user-specified CNI configuration files

### PR DESCRIPTION
**Pull request by squeed**
This lets us specify the exact CNI configuration file we want, but still only writes it when the network is ready.

### CODE REVIEW

**Comment by squeed** [comment]
/hold wip
This should be fine, but I haven't had a chance to test it. Plus, we need to tweak the MCD tempfiles.

**Comment by squeed** [comment]
/hold
After this merges, we'll updated the manifests in the network operator.

**Comment by squeed** [comment]
Bah, it failed tests because the file path is the readiness probe. Can't change that.

**Comment by squeed** [comment]
Test failures seem like a rebase issue. Race detector triggered on some upstream code.

**Comment by squeed** [comment]
Weird, ovs failed to come up in the last upgrade run. That shouldn't have anything to do with this PR, but it's unsettling...

File: pkg/cmd/openshift-sdn/cmd.go [hunk 1]
```diff
@@ -174,11 +188,10 @@ func (sdn *OpenShiftSDN) ValidateAndParse() error {
 		return err
 	}
 
-	cniConfDir := "/etc/cni/net.d"
-	if val, ok := sdn.NodeConfig.KubeletArguments["cni-conf-dir"]; ok && len(val) == 1 {
-		cniConfDir = val[0]
```
Comment (dcbw) [inline pkg/cmd/openshift-sdn/cmd.go]: @squeed so we're dropping the enforced same CNI configdir path between kubelet and the SDN and just making sure we manage that manually through the daemonset YAML now? Which is fine, just want to be sure.

Comment (squeed) [reply pkg/cmd/openshift-sdn/cmd.go]: basically, yes - given that NodeConfig doesn't exist anyways.

**Pull request closed without merging**